plasmapy/atomic/tests/test_atomic.py

```diff
@@ -48,352 +48,489 @@
+    nuclear_reaction_energy
 ]
 
+atomic_TypeError_badargs = [1.1, {'cats': 'bats'}, 1 + 1j]
 
-@pytest.mark.parametrize(
-    "argument, expected_error", mass_number_error_table)
-def test_mass_number_error(argument, expected_error):
-    """Test that `mass_number` raises the expected exceptions."""
-    with pytest.raises(expected_error):
-        mass_number(argument)
+atomic_ParticleErrors_funcs_table = [
+    atomic_symbol,
+    isotope_symbol,
+    atomic_number,
+    is_stable,
+    half_life,
+    mass_number,
+    element_name,
+    standard_atomic_weight,
+    isotope_mass,
+    ion_mass,
+    particle_mass,
+    known_isotopes,
+    stable_isotopes,
+    common_isotopes,
+    isotopic_abundance,
+    integer_charge,
+    electric_charge,
+]
 
+atomic_ParticleError_badargs = [
+    -1,
+    119,
+    'grumblemuffins',
+    'H-0',
+    'Og-294b',
+    'H-9343610',
+    'Fe 2+4',
+    'Fe+24',
+    'Fe +59',
+    'C++++++++++++++++',
+    'C-++++',
+    'h',
+    'd',
+    'he',
+    'au',
+    'alpha 1+',
+    'alpha-4',
+]
 
-# (argument, expected)
-element_name_table = [
-    ('D', 'hydrogen'),
-    ('deuterium', 'hydrogen'),
-    ('Au', 'gold'),
-    ('alpha', 'helium'),
-    ('helium-4', 'helium'),
-    ('H-2', 'hydrogen'),
-    ('Deuterium', 'hydrogen'),
-    ('Hydrogen-3', 'hydrogen'),
-    ('hydrogen-3', 'hydrogen'),
-    ('H-3', 'hydrogen'),
-    ('tritium', 'hydrogen'),
-    ('Alpha', 'helium'),
-    ('alpha', 'helium'),
-    (1, 'hydrogen'),
-    (26, 'iron'),
-    (79, 'gold'),
-    ('p', 'hydrogen'),
-    ('P', 'phosphorus'),
-    ('Be-8', 'beryllium'),
-    ('Li-7', 'lithium'),
-    ('N', 'nitrogen'),
-    ('N+++', 'nitrogen'),
-    ('D-', 'hydrogen')]
-
-
-@pytest.mark.parametrize("argument, expected", element_name_table)
-def test_element_name(argument, expected):
-    """Test that `element_name` returns the expected results."""
-    assert element_name(argument) == expected, \
-        (f"element_name({repr(argument)}) is returning a value of "
-         f"{element_name(argument)}, which differs from the expected "
-         f"value of {repr(expected)}.")
-
-
-# (argument, expected_error)
-element_name_error_table = [
-    ('vegancupcakes', InvalidParticleError),
-    ('C-+-', InvalidParticleError),
-    (1.24, TypeError),
-    ('n', InvalidElementError),
-    ('neutron', InvalidElementError),
-    (0, InvalidParticleError),
-    ('H++', InvalidParticleError),
-    ('t', InvalidParticleError),
-    ('pb', InvalidParticleError),
-    ('d', InvalidParticleError),
-    ('h-3', InvalidParticleError),
-    ('Pb-9', InvalidParticleError),
-    ('H 2+', InvalidParticleError),
+metatable = [
+    (atomic_TypeError_funcs_table, atomic_TypeError_badargs, TypeError),
+    (atomic_ParticleErrors_funcs_table, atomic_ParticleError_badargs, InvalidParticleError),
 ]
 
+for funcs, badargs, error in metatable:
+    for func in funcs:
+        for badarg in badargs:
+            all_tests += [[func, badarg, error]]
```

Comment:
Looking at this, I'm afraid I might not be able to find my way through any failing tests using this framework anymore...